### PR TITLE
Add hysteresis support for AMP gradient scale update

### PR DESCRIPTION
**Pull request by minitu**
This PR adds support for hysteresis in AMP gradient scale update.

### CODE REVIEW

File: tests/L0/run_amp/test_update_scale_hysteresis.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import unittest
+import random
+import math
+
+from apex import amp
```
Comment (crcrpar) [inline tests/L0/run_amp/test_update_scale_hysteresis.py:5]: Where is this used?

File: tests/L0/run_amp/test_update_scale_hysteresis.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  print("amp_C fused kernels unavailable, disabling TestUpdateScaleHysteresis.  ImportError was ", err)
+  disabled = True
+
+def isfinite(val):
+    return ((val >= torch.finfo(torch.float32).smallest_normal) and (val <= torch.finfo(torch.float32).max))
+
+class TestUpdateScaleHysteresis(unittest.TestCase):
+
+    def setUp(self):
+        common_init(self)
```
Comment (crcrpar) [inline tests/L0/run_amp/test_update_scale_hysteresis.py:24]: this doesn't sound necessary to me as per https://github.com/NVIDIA/apex/blob/2386a912164b0c5cfcd8be7a2b890fbac5607c82/tests/L0/run_amp/utils.py#L15

File: tests/L0/run_amp/test_update_scale_hysteresis.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import unittest
+import random
+import math
+
+from apex import amp
+import torch
+
+from utils import common_init
```
Comment (crcrpar) [inline tests/L0/run_amp/test_update_scale_hysteresis.py:8]: as `common_init` doesn't seem to have something to do with test logic.
```suggestion
```

**Comment by minitu** [comment]
@crcrpar Thanks, addressed your comments

**Review by crcrpar — APPROVED** [review]
